app/src/main/java/com/jtmcn/archwiki/viewer/WikiClient.java

```diff
@@ -1,44 +1,48 @@
 package com.jtmcn.archwiki.viewer;
 
-import android.os.AsyncTask;
-import android.os.Build;
+import android.util.Log;
 import android.webkit.WebView;
 import android.webkit.WebViewClient;
 import android.widget.Toast;
 
-import java.lang.ref.WeakReference;
+import com.jtmcn.archwiki.viewer.data.WikiPage;
+import com.jtmcn.archwiki.viewer.tasks.Fetch;
+import com.jtmcn.archwiki.viewer.tasks.FetchUrl;
+import com.jtmcn.archwiki.viewer.utils.AndroidUtils;
+
 import java.util.Stack;
 
-public class WikiClient extends WebViewClient {
+import static com.jtmcn.archwiki.viewer.Constants.ARCHWIKI_BASE;
+import static com.jtmcn.archwiki.viewer.Constants.TEXT_HTML_MIME;
+import static com.jtmcn.archwiki.viewer.Constants.UTF_8;
 
-	protected static WeakReference<WebView> wrWeb;
-	private static boolean pageFinished;
-	private static WikiPage webpage;
-	private static Stack<String> histHtmlStack = new Stack<>();
-	private static Stack<String> histTitleStack = new Stack<>();
-	private String myUrl;
-	private String pageTitle;
+public class WikiClient extends WebViewClient implements FetchUrl.OnFinish<WikiPage> {
+	public static final String TAG = WikiClient.class.getSimpleName();
+	private WebView webView;
+	private boolean pageFinished;
+	private Stack<WikiPage> webpageStack = new Stack<>();
 
 	public WikiClient(WebView wikiViewer) {
-		wrWeb = new WeakReference<>(wikiViewer);
+		webView = wikiViewer;
 	}
 
 	/*
 	 * Manage page history
 	 */
-	public static void addHistory(String histHtml, String histTitle) {
-		histHtmlStack.push(histHtml);
-		histTitleStack.push(histTitle);
+	public void addHistory(WikiPage wikiPage) {
+		webpageStack.push(wikiPage);
+		Log.d(TAG, "Adding page " + wikiPage.getPageTitle() + ". Stack size now at " + webpageStack.size());
 	}
 
-	public static void loadWikiHtml(String wikiHtml) {
-		String urlStr = "https://wiki.archlinux.org/";
-		String mimeType = "text/html";
-		String encoding = "UTF-8";
-
+	public void loadWikiHtml(WikiPage wikiPage) {
 		// load the page in webview
-		wrWeb.get().loadDataWithBaseURL(urlStr, wikiHtml, mimeType, encoding,
-				null);
+		webView.loadDataWithBaseURL(
```

Comment:
I think this is the cause of #22 since it's just loading the html and not using the anchor from the url and scrolling to that section